### ISSUE TITLE
Property not found on index accessed type in a union

### ISSUE THREAD

**Issue by sandersn** [Microsoft/TypeScript#21464]
```ts
interface I {
    [key: string]: I
    (...args: any[]): I
}

declare let y: string | I
y.toUpperCase() // Error
```

Expected: No error: both `string` and `I` separately allow property access of `toUpperCase`.

Actual: Error, "Property 'toUpperCase' does not exist on type 'I'.

**Comment by ahejlsberg** [comment]
Less confusing repro:

```ts
interface A {
    a: string;
}
interface I {
    [key: string]: number;
}

declare let y: A | I
y.a;  // Got error, expected string | number
```

**Comment by mhegazy** [comment]
Looks like a duplicate of https://github.com/Microsoft/TypeScript/issues/21141
